specification.md

```diff
@@ -0,0 +1,1243 @@
+
+Inline data (data in JSON format within the Descriptor) must not be used. This means that a `path` must be provided for the Resource. The Resource path shall point to a file reference (for packages stored on disk) or a URL.
+
+The `schema` property of the resource must be provided inline, and must not use an external schema file or URL.
+
+The `schema` property must contain a `fields` object describing the 6 columns within the Resource data. These fields are common to all Flow Results Packages, but are provided here for compatibility with software designed to dynamically read Tabular Data Resources:
+
+```
+{
+  "fields":[
+    {
+      "name":"timestamp",
+      "title":"Timestamp",
+      "type":"datetime"
+    },
+    {
+      "name":"row_id",
+      "title":"Row ID",
+      "type":"string"
+    },
+    {
+      "name":"contact_id",
+      "title":"Contact ID",
+      "type":"string"
+    },
+    {
+      "name":"question_id",
+      "title":"Question ID",
+      "type":"string"
+    },
+    {
+      "name":"response",
+      "title":"Response",
+      "type":"any"
+    },
+    {
+      "name":"response_metadata",
+      "title":"Response Metadata",
+      "type":"object"
+    }
+  ]
+}
+```
+
+The `schema` property must additionally contain a `questions` object describing metadata for all the Questions pertaining to Responses in this package.  The object identifier (e.g.: 'ae54d3') of questions in this object connects to the Question ID found in each Response row:
+
+```
+{
+  "questions":{
+    "ae54d3":{
+      "type":"multiple_choice",
+      "label":"Are you male or female?",
+      "type_options":{
+        "choices":[
+          "male",
+          "female",
+          "not identified"
+        ]
+      }
+    },
+    "ae54d7":{
+      "type":"multiple_choice",
+      "label":"Favorite ice cream flavor?",
+      "type_options":{
+        "choices":[
+          "chocolate",
+          "vanilla",
+          "strawberry"
+        ]
+      }
+    },
+    "ae54d8":{
+      "type":"numeric",
+      "label":"How much do you weigh, in lbs?",
+      "type_options":{
+        "range":[1, 350]
+      }
+    },
+    "ae54da":{
+      "type":"open",
+      "label":"How are you feeling today?",
+      "type_options":{
+
+      }
+    }
+  }
+}
+```
+
+The following properties are required for each question:
+
+<table>
+  <tr>
+    <td>Property</td>
+    <td>Description</td>
+    <td>Example</td>
+  </tr>
+  <tr>
+    <td>`type`</td>
+    <td>Describes the semantic type of the Question, which must be from the following list:
+multiple_choice_one
+multiple_choice_many
+numeric
+open
+text
+image
+video
+audio
+geo_point
+date
+time
+datetime</td>
+    <td>'type':'multiple_choice_one'</td>
+  </tr>
+  <tr>
+    <td>`label`</td>
+    <td>A human-readable label that can be used to present and provide context for this Question/Response.  This is provided in a single default language; localization is left outside the scope of this specification.</td>
```

Comment:
Since localization is out of scope, shouldn't the default language of the schema be specified somewhere?